diff --git a/include/FramelessHelper/Core/private/framelesshelpercore_global_p.h b/include/FramelessHelper/Core/private/framelesshelpercore_global_p.h
index 26ad76a..616f662 100644
--- a/include/FramelessHelper/Core/private/framelesshelpercore_global_p.h
+++ b/include/FramelessHelper/Core/private/framelesshelpercore_global_p.h
@@ -167,15 +167,15 @@ FRAMELESSHELPER_END_NAMESPACE
   } \
   [[maybe_unused]] [[nodiscard]] inline constexpr bool operator>=(const Type1 &lhs, const Type2 &rhs) noexcept \
   { \
-      return (operator>(lhs, rhs) || operator==(lhs, rhs)); \
+      return ((lhs.width() * lhs.height()) >= (rhs.width() * rhs.height())); \
   } \
   [[maybe_unused]] [[nodiscard]] inline constexpr bool operator<(const Type1 &lhs, const Type2 &rhs) noexcept \
   { \
-      return (operator!=(lhs, rhs) && !operator>(lhs, rhs)); \
+      return ((lhs.width() * lhs.height()) < (rhs.width() * rhs.height())); \
   } \
   [[maybe_unused]] [[nodiscard]] inline constexpr bool operator<=(const Type1 &lhs, const Type2 &rhs) noexcept \
   { \
-      return (operator<(lhs, rhs) || operator==(lhs, rhs)); \
+      return ((lhs.width() * lhs.height()) <= (rhs.width() * rhs.height())); \
   }
 
 DECLARE_SIZE_COMPARE_OPERATORS(QSize, QSize)
